### ISSUE TITLE
Require system auth on tapping show nsec

### ISSUE THREAD

**Issue by alltheseas** [0xchat-app/0xchat-core#5]
![image](https://github.com/0xchat-app/0xchat-core/assets/64376233/ee0a393f-09c3-4266-958a-5c9b0a9ebc47)

_suggestion_

Hide nsec behind iOS system auth when user taps on the show nsec button. 

Do not show nsec is auth not provided. 

Show nsec only on successful system auth.

**Comment by water783** [comment]
Thx, will add it to our plans

**Comment by water783** [comment]
This feature is included in the v1.2.3 release

**Comment by alltheseas** [comment]
I am on 1.2.3 (16) and cannot confirm this feature is on 1.2.3 (16).

**Comment by water783** [comment]
Do you have turned on the passcode & FaceID 

![image](https://github.com/0xchat-app/0xchat-core/assets/130329555/246c7ee7-a79f-4a91-9057-09cac459efcf)

**Comment by alltheseas** [comment]
I see what changes you made now. 

You added a custom pw for 0xchat. My original suggestion was to use the system auth (i.e. iOS pw). This is fine. Telegram uses this pattern. 

Confirming after I enable pw, 0xchat asks me for login to view the nsec. 

However I can copy the nsec without auth, which reduces the value of see nsec auth. 

I'll add a new ticket for this.

**Comment by water783** [comment]
1. Emm, the idea behind our design is to set a password for 0xchat to enhance the security of Auto-login. If users find it inconvenient, they can enable Face ID for quick access. However, this is contingent on user confirmation. Using the iOS password is also a good choice, but it's possible that the user might not have set up an iOS password. 

2. Being able to copy nsec without auth is a mistake, will fix it. :)
